hibernate-core/src/main/java/org/hibernate/engine/spi/ActionQueue.java

```diff
@@ -1189,9 +1189,11 @@ public void sort(List<AbstractEntityInsertAction> insertions) {
 
 					if ( nextBatchIdentifier.hasAnyParentEntityNames( batchIdentifier ) ) {
 						nextBatchIdentifier.parent = batchIdentifier;
+						nextBatchIdentifier.getParentEntityNames().add(batchIdentifier.getEntityName());
 					}
 					if ( batchIdentifier.hasAnyChildEntityNames( nextBatchIdentifier ) ) {
 						nextBatchIdentifier.parent = batchIdentifier;
+						nextBatchIdentifier.getParentEntityNames().add(batchIdentifier.getEntityName());
```

Comment:
Please refer to https://github.com/hibernate/hibernate-ide-codestyles on how to make your code style consistent with existing one. Especially, empty chars are added aggressively to help code reading. For instance
```
nextBatchIdentifier.getParentEntityNames().add(batchIdentifier.getEntityName());
``` 
should be re-formatted as
```
nextBatchIdentifier.getParentEntityNames().add( batchIdentifier.getEntityName() );
```

Reply:
updated